internal/server/handlers/config_routine.go

```diff
@@ -303,14 +303,14 @@ func (s *Service) DisableRoutine(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, routineNameNotSpecifiedMsg, http.StatusBadRequest)
 		return
 	}
-	handler, found := s.handlerHolder.Load(routineName)
+	_, found := s.handlerHolder.Load(routineName)
 	if !found {
 		hLogger.Error("unknown routine name", slog.String("name", routineName))
 		http.Error(w, fmt.Sprintf("Routine %s could not be found", routineName), http.StatusNotFound)
 		return
 	}
 
-	handler.Cancel() // cancel any running job for this routine before disabling it.
+	s.registry.Cancel(routineName) // cancel any running job for this routine before disabling it.
```

Comment:
Should be canceled after updating the config to avoid a race condition where a backup may start between operations.